pydantic/json_schema.py

```diff
@@ -0,0 +1,1081 @@
+from __future__ import annotations
+
+import math
+import re
+from dataclasses import is_dataclass
+from enum import Enum
+from ipaddress import IPv4Address, IPv4Interface, IPv4Network, IPv6Address, IPv6Interface, IPv6Network
+from pathlib import PurePath
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Counter,
+    Dict,
+    List,
+    NewType,
+    Optional,
+    Pattern,
+    Sequence,
+    Type,
+    Union,
+    cast,
+)
+from uuid import UUID
+
+from pydantic_core import CoreSchema, CoreSchemaType, core_schema
+from pydantic_core.core_schema import TypedDictField
+from typing_extensions import Literal, TypedDict, TypeGuard
+
+from ._internal._core_metadata import CoreMetadataHandler
+from ._internal._typing_extra import all_literal_values, is_namedtuple
+from ._internal._utils import get_model, lenient_issubclass
+from .errors import PydanticInvalidForJsonSchema, PydanticUserError
+from .networks import IPvAnyAddress, IPvAnyInterface, IPvAnyNetwork
+
+if TYPE_CHECKING:
+    from .dataclasses import Dataclass
+    from .main import BaseModel
+
+JsonSchemaValue = Dict[str, Any]
+
+# ##### JSON Schema Metadata Manipulation #####
+# Keys missing in a JsonSchemaMetadata should be treated the same as they would be if present with a value of None.
+# This is important to ensure that it is possible to "remove" an override by setting it to None.
+JsonSchemaMetadata = TypedDict(
```

Comment:
is there a reason this can't be defined using the class syntax for `TypedDict`?

Reply:
The '$comment' key. We can drop it, but doing it like this allowed me to skip some key-remapping logic. If there is a way to use the class syntax with a different key name I would prefer that (like alias for pydantic model fields), I just couldn't find such a thing.

Reply:
okay, keep.